src/main/host/descriptor/regular_file.c

```diff
@@ -381,6 +450,23 @@ ssize_t regularfile_preadv(RegularFile* file, const Host* host, const struct iov
         return (ssize_t)_regularfile_readvRandomBytes(file, host, iov, iovcnt);
     }
 
+    if (file->type == FILE_TYPE_IN_MEMORY) {
+        if (file->inMemoryFile.content == NULL)
+            return -EBADF;
+        ssize_t read = 0;
+        for (int i = 0; i < iovcnt; i++) {
+            ssize_t left = file->inMemoryFile.contentLen - offset;
+            if (!left)
+                break;
+            ssize_t to_read = MIN(left, iov[i].iov_len);
+            memcpy(iov[i].iov_base, file->inMemoryFile.content + offset, to_read);
```

Comment:
Should check that `iov_base` is not NULL (return an error if NULL and `iov_len` is non-zero, skip if NULL and `iov_len` is zero).